packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -122,12 +123,13 @@ class WebpackCLI extends GroupHelper {
          * options is an array (multiple configuration) so we create a new
          * configuration where each element is individually merged
          */
+        // console.log({ options, strategy });
         if (Array.isArray(options)) {
             this.compilerConfiguration = options.map((configuration) => {
                 if (strategy) {
                     return webpackMerge.strategy(strategy)(this.compilerConfiguration, configuration);
                 }
-                return webpackMerge(this.compilerConfiguration, configuration);
+                return webpackMerge(configuration, this.compilerConfiguration);
```

Comment:
I think we should revert this, not quite sure why I made this change 😅

Reply:
But this seems logically incorrect, configuration should take precedence

Reply:
Let's keep this change.

Reply:
Did you confirm?

Reply:
well it's not necessary to write it like this, but I see the `commonConfig` is passed as first argument in docs.

Reply:
Do we have tests on this change?

Reply:
No but I see this change is not affecting anything, can be safely reverted.

Reply:
Let's add tests on this

Reply:
how should we test it.?

Reply:
Need investigate, right now we have:
```
return webpackMerge.strategy(strategy)(this.compilerConfiguration, configuration);
```

and

```
return webpackMerge(configuration, this.compilerConfiguration);
```

I think here bug

Reply:
@anshumanv any suggestions?

Reply:
Yep let's revert this for now, I think I changed while debugging, we can revisit later to add tests